mypy/stubtest.py

```diff
@@ -914,7 +914,8 @@ def _verify_signature(
     elif len(stub.pos) < len(runtime.pos):
         for runtime_arg in runtime.pos[len(stub.pos) :]:
             if runtime_arg.name not in stub.kwonly:
-                yield f'stub does not have argument "{runtime_arg.name}"'
+                if not _is_private_parameter(runtime_arg):
+                    yield f'stub does not have argument "{runtime_arg.name}"'
```

Comment:
What about something like:

```py
# RUNTIME
def foo(a, b, _c, d): pass
```

```py
# STUB
def foo(a, b, d) -> None: ...
```

We should probably still emit an error for something like that, right?

What about something like this? (The diff is relative to your PR branch):

```diff
diff --git a/mypy/stubtest.py b/mypy/stubtest.py
index f47dbb2fd..91c84fe58 100644
--- a/mypy/stubtest.py
+++ b/mypy/stubtest.py
@@ -912,9 +912,10 @@ def _verify_signature(
             if stub.varpos is not None:
                 yield f'runtime does not have *args argument "{stub.varpos.variable.name}"'
     elif len(stub.pos) < len(runtime.pos):
-        for runtime_arg in runtime.pos[len(stub.pos) :]:
+        params_not_in_stub = runtime.pos[len(stub.pos) :]
+        for i, runtime_arg in enumerate(params_not_in_stub):
             if runtime_arg.name not in stub.kwonly:
-                if not _is_private_parameter(runtime_arg):
+                if not all(_is_private_parameter(param) for param in params_not_in_stub[i:]):
                     yield f'stub does not have argument "{runtime_arg.name}"'
             else:
                 yield f'runtime argument "{runtime_arg.name}" is not keyword-only'
```

Reply:
I didn't want to get too bogged down by special cases. When a positional parameter is not at the end of the positional (or positional-or-kwargs) list, it's ambiguous whether it's truly a private parameter, as you have to either fill it or use kwargs syntax for the following arguments. So this example:

```python
def foo(a, b, _c = None, d = None): pass
```

would need to be annotated as

```python
def foo(a, b, *, d = None): pass
```

I feel this is a bit outside stubtest's pay level. But I added a test for this.

Reply:
fair enough!